woocommerce/woocommerce
# [Settings] Add checkbox, input, and select

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Add checkbox, input, and select components from `@wordpress/components`. I decided to keep state internal to each component because we'll need to gather values from all inputs including any inputs custom HTML may provide.

<img width="902" alt="image" src="https://github.com/user-attachments/assets/6c1de64f-b91f-4db9-adf2-1881b872de30" />




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add/Enable Gutenberg and turn on the `settings` feature flag.
2. WooCommerce > Settings.
3. Poke around and see checkboxes, selects, and inputs rendered with values.
4. Compare to LSwxXIhP15A1YvZWnkgihH-fi-1832-125630 but we really don't have definitive designs at this time so I left things as unstyled as possible going with the baseline Gutenberg styles as is.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

Add Settings checkbox, input, and select

</details>
